guides/micronaut-cloud-database-oracle/micronaut-cloud-database-oracle.adoc

```diff
@@ -0,0 +1,386 @@
+"port": 3306,
+"port-x": 33060
+----
+
+external:micronaut-cloud-database-base/create-app.adoc[]
+
+== Deploying the Application
+
+Use the guideLink:micronaut-oracle-cloud[Deploy a Micronaut application to Oracle Cloud] guide to create a compute instance and deploy the application to it; follow the steps in the "Create an Oracle Cloud Compute Instance" and "Deploy to Oracle Cloud" sections up to the step where you start the application. We need to connect the application to the MySQL database before starting it up.
+
+TODO ingress rule
+
+NOTE: When creating the compute VM at Oracle Cloud, use the same subnet as the one where you created the MySQL database, otherwise the application will not be able to access the database.
+
+== Configure MySQL
+
+Flyway will create the database tables for us the first time the application starts, but we must create the database and a database user first.
+
+There are no direct ways to externally https://docs.oracle.com/en-us/iaas/mysql-database/doc/connecting-db-system.html[connect to a MySQL database] in Oracle Cloud, so we'll do the work from the compute instance.
+
+=== MySQL client
+
+Install the MySQL client:
+
+[source,bash]
+----
+sudo yum install mysql
+----
+
+Connect to MySQL with the admin username and password you chose earlier and with the MySQL IP address:
+
+[source,bash]
+----
+mysql --host <MySQL_IPAddress> -u <admin_username> -p
+----
+
+=== Create a database and user
+
+Create the database (use any valid database name, e.g. `micronaut`):
+
+[source,mysql]
+----
+mysql> CREATE DATABASE micronaut;
+----
+
+Create a database user (use any valid MySQL username, e.g. `guide_user`, and a valid password):
+
+[source,mysql]
+----
+mysql> CREATE USER 'guide_user'@'<compute instance private IP address>' IDENTIFIED BY <user password>;
+----
+
+Grant access to the database for the new user:
+
+[source,mysql]
+----
+mysql> GRANT ALL ON micronaut.* TO 'guide_user'@'<compute instance private IP address>';
+----
+
+Exit the MySQL console:
+
+[source,mysql]
+----
+mysql> exit
+----
+
+== MySQL Environment Vars
+
+Create environment variables for `JDBC_URL`, `JDBC_USER`, `JDBC_PASSWORD` which will be used in the Micronaut app's `application.yml` datasource:
+
+[source,bash]
+----
+export JDBC_URL=jdbc:mysql://<MySQL IP address>:3306/micronaut
+export JDBC_USER=guide_user
+export JDBC_PASSWORD=<user password>
+----
+
+TODO curl examples to cloud compute instance
+
+== Next steps
+
+When you are finished using the database you can https://docs.oracle.com/en-us/iaas/tools/oci-cli/2.10.4/oci_cli_docs/cmdref/mysql/db-system/delete.html[delete it using the CLI]. Run
+
+[source,bash]
+----
+oci mysql db-system delete \
+    --db-system-id ocid1.mysqldbsystem.oc1.iad.aaaaaaaa2pq3a37hftut...
+----
+
+The output should look like this:
+
+[source,json]
+----
+{
+"opc-work-request-id": "ocid1.mysqlworkrequest.oc1.iad.e68d5dc7-92be-45..."
+}
+----
+
+external:micronaut-cloud-database-base/end.adoc[]
```

Comment:
I don't understand the benefit of using this instead of a common snippet. Why did we needed `external`. Could not we just have?

```
common:micronaut-cloud-database-end.adoc

```

Reply:
Resolved via https://github.com/micronaut-projects/micronaut-guides/pull/743